File: javascript/grains/grains.js
const bigInt = require('big-integer');

const totalGrains = bigInt(2).pow(64).minus(1).toString();

const Grains = function Grains() {
  this.square = function square(n) {
    return bigInt(2).pow(n - 1).toString();
  };

  this.total = function total() {
    return totalGrains;
  };
};

module.exports = Grains;

File: haskell/grains/src/Grains.hs
module Grains (square, total) where

import Data.Maybe (fromJust)

square :: Integer -> Maybe Integer
square n = if 0 < n && n <= 64
           then Just $ 2 ^ (n - 1)
           else Nothing

total :: Integer
total = sum $ map (fromJust . square) [1..64]